campaignion_tracking/campaignion_tracking.module

```diff
@@ -13,3 +22,193 @@ function campaignion_tracking_preprocess_html(&$vars) {
+  $is_donation = $type && $type->isDonation();
+  $webform = Webform::fromNode($form['#node']);
+
+  // Information about the current webform.
+  drupal_add_js([
+    'campaignion_tracking' => [
+      'context' => [
+        'webform' => [
+          'total_steps' => $form_state['webform']['page_count'],
+          'current_step' => $form_state['webform']['page_num'],
+          'last_completed_step' => isset($form_state['webform']['page_visited']) ? $form_state['webform']['page_visited'] : 0,
+        ],
+      ],
+    ],
+  ], 'setting');
+
+  // Information about a donation (in case the node is one).
+  if ($is_donation) {
+    $paymethod = $webform->componentByKey('paymethod_select');
+    $currency_code = isset($paymethod['extra']['currency_code']) ? $paymethod['extra']['currency_code'] : 'EUR';
+    $submission = $webform->formStateToSubmission($form_state);
+
+    // NB: Only after the first step submit we get values.
+    // Thus `interval` and `amount` might be "null" in JS (in case no values
+    // have been submitted so far)
+    drupal_add_js([
+      'campaignion_tracking' => [
+        'context' => [
+          'donation' => [
+            'currency_code' => $currency_code,
+            'interval' => $submission->valueByKey('donation_interval'),
+            'amount' => $submission->valueByKey('donation_amount'),
+          ],
+        ],
+      ],
+    ], 'setting');
+  }
+
+  if ($is_donation) {
+    $form['#submit'][] = 'campaignion_tracking_add_donation_success_fragment';
+  }
+}
+
+/**
+ * Submit-callback for webform_client_form().
+ *
+ * Add donation success tracking data to the URL-fragment of the redirect.
+ */
+function campaignion_tracking_add_donation_success_fragment($form, &$form_state) {
+  if (empty($form_state['webform_completed'])) {
+    return;
+  }
+  // TODO: extend later for added trackers.
+  if (module_exists('campaignion_tracking_gtm')) {
+    $fragment = 'gtm=ds';
+    if (!empty($form_state['redirect'][1]['fragment'])) {
+      $form_state['redirect'][1]['fragment'] .= ';' . $fragment;
+    }
+    else {
+      $form_state['redirect'][1]['fragment'] = $fragment;
+    }
+  }
+}
+
+/**
+ * Load a payment.
+ *
+ * TODO: is there a little_helper for that?
+ */
+function _campaignion_tracking_load_payment($nid, $sid) {
+  $subquery = db_select('campaignion_activity_webform', 'w')
+    ->fields('w', ['activity_id'])
+    ->condition('nid', $nid)
+    ->condition('sid', $sid);
+  $pid = db_select('campaignion_activity_payment', 'p')
+    ->fields('p', ['pid'])
+    ->condition('p.activity_id', $subquery, 'in')
+    ->execute()
+    ->fetchColumn();
+  if ($pid) {
+    $payments = entity_load('payment', [$pid]);
+    return array_shift($payments);
+  } else {
+    return FALSE;
+  }
+}
+
+/**
+ * Implements hook_node_view().
+ *
+ * Information from the previous node, where a webform was submitted.
+ *
+ * TODO: Maybe also do this in node_view
+ */
+function campaignion_tracking_page_build(&$page) {
```

Comment:
Shouldn’t this be made obsolete by the fragment appended to the redirect URL?

Reply:
gone now.

the fragment triggers a message on the trackermanager and the gtm tracker uses it's sessionStorage state to get the information (along with additional information from the fragment like the sid and paymethod)